vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1295,6 +1300,31 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private int getNumChildrenWithType(Node parent, Node child) throws RepositoryException {
+
+        int count=0;
+       
```

Comment:
The idea is:
- get the `NodeDefinition` of the child node that is to be deleted
- loop through all childnodes and count the ones that have the same nodedef
- if the nodedef is mandatory and there are more than 1 nodes with this nodedef, we can delete it.